src/command-emulation.ts

```diff
@@ -0,0 +1,85 @@
+import fs from 'fs'
+import path from 'path'
+import { promisify } from 'util'
+
+const unlinkAsync = promisify(fs.unlink)
+const rmdirAsync = promisify(fs.rmdir)
+const writeFileAsync = promisify(fs.writeFile)
+const chmodAsync = promisify(fs.chmod)
+
+import { createTempDirectory } from './common'
+
+interface CommandEmulationOptions {
+  overridePath: boolean
+}
+
+export class CommandEmulation {
+  public tmpdir!: string
+  private oldPath!: string
+  private newPath!: string
+  private commands: string[] = []
+  private initPromise: Promise<void>
+
+  constructor(options?: CommandEmulationOptions) {
+    this.initPromise = this.init(options)
+    this.initPromise.catch(() => {
+      // Do nothing as we will throw at later calls
+    })
+  }
+
+  static async new(options?: CommandEmulationOptions): Promise<CommandEmulation> {
```

Comment:
How does this work? Is that just to do
```
const commandEmulation = await CommandEmulation.new()
```
instead of
```
const commandEmulation = new CommandEmulation()
await commandEmulation.setup()
```